knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,60 @@
+
+## Decision 2
+
+| Current Name, Aliases                                    | Future Name, Aliases             | Current Behavior                                                                                  | Future Behavior                                                                                                                                              | Allow-no |
+| :------------------------------------------------------- | :------------------------------- | :------------------------------------------------------------------------------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------- | :------: |
+| `--inputDir`, `-i` _(required)_                          | `--input`, `-i` _(required)_     | Input directory/file                                                                              | _same_                                                                                                                                                       |  false   |
+| `--outputDir`, `-o` _(required)_                         | `--outputDir`, `-o` _(required)_ | Output directory                                                                                  | _same_                                                                                                                                                       |  false   |
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                       |  false   |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. Can only be set if tsconfig is enabled. This should be explicitly stated in the documentation.                                    |  false   |
+| `--generatePackageJson`                                  | `--packageJson`                  | Writes a default package.json, default true.                                                      | Writes a default package.json, default true (needs core dependency). Optionally in the future: Writes a custom package.json if passed. Keep boolean for now. |   true   |
+| `--serviceMapping`                                       | `--serviceMapping`               | Considers a service mapping file. Shows a warning if none is given (differs from OData behavior). | Changes the service mapping path. Generates it always.                                                        |   false   |
```

Comment:
@FrankEssenberger @jjtang1985 
I updated this according to [this comment](https://github.com/SAP/cloud-sdk-js/pull/1218#discussion_r615884737).

Reply:
I do not get this part `Changes the service mapping path. Generates it always.`

Reply:
I thought that if not given it will generate the file and use it in a potential next run so there must be a default path where the generator looks for the mapping. This should be mentioned in the description.

Reply:
Not sure about the `Changes the service mapping path. Generates it always.`
Here is my proposal:
Is this flag has value?
- Y.
  - Does the file exist with the path?
    - Y
      - Use the provided file.
    - N
      - Generate a default file with the given path.
- N.
  - Use default path for the service mapping, generate a default file.

Reply:
@jjtang1985's tree is correct. This parameter is just there to provide a custom path, I will improve the wording a bit. The texts in this ADR are not the descriptions of the parameters as the users would see it.